specification/flag-evaluation/flag-evaluation.md

```diff
@@ -192,3 +190,9 @@ Implementations may define a standard logging interface that can be supplied as
 > The `client` **SHOULD** provide asynchronous or non-blocking mechanisms for flag evaluation.
 
 It's recommended to provide non-blocking mechanisms for flag evaluation, particularly in languages or environments wherein there's a single thread of execution.
+
+##### Requirement 1.22
+
+> The `client` **MUST** transform the `evaluation context` using the `provider's` `context transformer` function, before passing the result of the transformation to the provider's flag resolution functions.
```

Comment:
The client actually RUNS the transformer (by default a no-op), before passing that transformed context to the provider's resolution methods. That's what I was trying to capture here.

Reply:
I think it's an unnecessary requirement. It's implicit that the provider must transform the content.

Reply:
@agentgonzo @justinabrahms 

> I think it's an unnecessary requirement. It's implicit that the provider must transform the content.

The provider must transform the context, yes, but if a context transform is defined, it's actually not the provider that calls that method. See the example for the [launchdarkly experimental provider](https://github.com/open-feature/sdk-research/blob/main/packages/js-launchdarkly-provider/src/lib/js-launchdarkly-provider.ts#L26-L34)

Here a context transformer is defined. It just sets a user ID, and puts all the custom attributes into a `custom` field, as the LaunchDarkly API dictates. Though the provider defines this function, it never runs it. The caller (the OpenFeature client) actually does before calling the flag resolution methods. This is nice, because you can see the flag resolution methods now take the transformed context, with the correct type as the parameter (`LDUser` supplied by the generic arg to the class): https://github.com/open-feature/sdk-research/blob/main/packages/js-launchdarkly-provider/src/lib/js-launchdarkly-provider.ts#L71-L88

This gives us a bit type safety through generics, and avoids calling transform code in every flag resolution method.

Reply:
I think it's nice for us to give something like the transformer to help simplify the provider implementation, but I also don't think it should be part of the spec. It really feels like more of an implementation detail. Maybe there's something we could put into the spec that's defines this as a suggested patterns for people implementing a client.

Reply:
I like a may. That's exactly what may is for. "Hey, we think this might be a good idea. Why don't you do that if you want." Either works for me though.

Reply:
I've made everything related to the context transformer a "MAY".